src/features/hdr-tonemap.md

```diff
@@ -0,0 +1,85 @@
+# HDR, Tonemapping, Bloom, Dithering
+
+{{#include ../include/links.md}}
+
+HDR (High Dynamic Range) refers to the ability of the game engine to handle very
+bright lights or colors. Bevy supports HDR rendering. This means you can have
+objects with colors that go above `1.0`, very bright lights, or bright emissive
+materials. All of this is supported for both 3D and 2D.
+
+This is not to be confused with HDR display output, which is the ability to
+produce a HDR image to be displayed by a modern monitor or TV with HDR
+capabilities. Bevy has no support for this yet.
+
+## Camera HDR configuration
+
+There is a per-camera toggle that lets you decide whether you want Bevy to
+store the HDR data internally during rendering.
+
+```rust,no_run,noplayground
+{{#include ../code/src/features/camera/hdr.rs:hdr-config}}
+```
+
+If it is enabled, Bevy's intermediate textures will be in HDR format. This
+allows you to enable post-processing effects like Bloom, that make use of the
+HDR rendering data. Tonemapping will happen as a post-processing step.
+
+If it is disabled, the shaders are expected to do any [tonemapping][cb::tonemap]
+internally, and output standard RGB colors. The "physically based" materials can
+still handle bright lights and other "HDR" scenarios, but that information will
+not be stored, and effects like Bloom will not work.
```

Comment:
I don't reaaally like the wording "that information will not be stored". My understanding is that the main pass can result in full-range linear RGB values that can be well beyond the limited range of 0.0 to 1.0. However, as bevy does not yet support outputting full-range / HDR data to displays, tonemapping is used to compress the full range back into the 0.0 to 1.0 range. Information is lost due to limited precision and compression of brighter values, but the point of tonemapping is to enable representation of those very bright full range values within a standard definition display code path as we have in bevy today.

I don't really like the way we call the option 'hdr: bool' when it really just means that the main pass outputs full range linear RGB values in a full range float texture, and tonemapping is done separately, rather than before writing to the 0.0-1.0 range texture.

Reply:
Yes, exactly. That's what I was trying to describe. `hdr = false` means that the shader has to output SDR (the hdr is not "stored"/preserved) and so tonemapping is done in-shader. `hdr = true` means that the shader can output HDR (hdr is stored/preserved), and tonemapping is done later, allowing for things like bloom.

What better wording can you suggest?

Reply:
I reworded it. I think it is better now. I'm going to publish it as it is now. If you still don't like it, feel free to respond, and I can make more improvements later.